### ISSUE TITLE
Crash in Bionics menu

### ISSUE THREAD

**Issue by Waladil** [CleverRaven/Cataclysm-DDA#13536]
Reported from Windows build 3665. None of the descriptions of updates between this and the current version (3670) seem to be a fix to this bug.

Currently, the bionics menu allows you to activate active bionics if you're on the passive tab, and attempt to activate passive bionics (showing the 'cannot be activated' message) from the active tab. If you do either of those things then press ! to show bionic descriptions, then the game crashes. I believe this to be because activating/attempting to activate a bionic then showing descriptions causes the most-recently-activated bionic to show its description.
Fixes (pick one):
1. Prevent cross-tab use of bionics.
2. Remove code that causes the most-recently-activated-bionic to show its description.
3. Allow the game to switch tabs.
4. If the most-recently-activated bionic is on the other tab, default to the first item on the current tab.

**Comment by namarrgon** [comment]
I can reproduce it on 0.C-5992-g3001c6d.

```
#0  0x00007ffff66595f8 in raise () from /usr/lib/libc.so.6
#1  0x00007ffff665aa7a in abort () from /usr/lib/libc.so.6
#2  0x00007ffff6f94595 in __gnu_debug::_Error_formatter::_M_error (this=0x7fffffffbdd0)
    at /build/gcc-multilib/src/gcc-5.2.0/libstdc++-v3/src/c++11/debug.cc:782
#3  0x000000000067e0ed in std::__debug::vector<bionic*, std::allocator<bionic*> >::operator[] (this=0x7fffffffc520, 
    __n=17) at /usr/include/c++/5.2.0/debug/vector:406
#4  0x000000000066f0b0 in player::power_bionics (this=0x59e5bf0) at src/bionics.cpp:418
#5  0x000000000057e03d in game::handle_action (this=0x50c0b30) at src/game.cpp:2697
#6  0x0000000000571bf4 in game::do_turn (this=0x50c0b30) at src/game.cpp:1258
#7  0x00000000004a79cc in main (argc=0, argv=0x7fffffffe300) at src/main.cpp:439
```

Full backtrace: https://gist.github.com/namarrgon/b93ca780c97c02346ec0

**Comment by drbig** [comment]
@Waladil: You mean like the bionic shortcuts work for all bionics (regardless of tab). E.g. 'a, b, c' are active and 'd, e' are passive, and then on active bionics tab pressing 'd' tries to activate the passive bionic?

If so then I definitely choose:
> 1. Prevent cross-tab use of bionics.

**Comment by Waladil** [comment]
Basically, yes, drbig. It can also lead to activating, I dunno, sonic resonator while on the passive tab. And then it does fully activate. Personally 1's the fix I would go with as well. Give it a whirl! Just... make sure you don't accidentally use a really bad bionic. Or press ! after, unless you've got a recent save too.

**Comment by BevapDin** [comment]
Fixed by #13551.

> Prevent cross-tab use of bionics.

If this is going to happen, the key bindings of the bionics should be exclusive for each tab (have an active bionic with key 'a' *and* a passive one with key 'a'). Because at that point, there is no reason to not allow that.
